Multiple target frameworks

zhuxb711 opened the pull request:
Previously, only .Net 7.0 is supported only, and most C# repo might not be able to use that.
This PR add support for multiple target framework: `net48` `netstandard2.0` `net6.0` `net7.0`

zhuxb711 committed:
Init commit

zhuxb711 committed:
Fixed Package Reference

zhuxb711 committed:
Update build.yml

zhuxb711 commented:
@noamyogev84 Hi Could take a look at this and publish to nuget if possible? Thanks very much!

noamyogev84 commented:
Heya @zhuxb711 that's really great!
Would you add to the README something about our multi target support now? it's worth mentioning it!

zhuxb711 committed:
Update README.md

zhuxb711 commented:
@noamyogev84 Thanks and README.md is updated to mention multi-target now

zhuxb711 committed:
Other improvements

1. Mark the test project with <IsTestProject>true</IsTestProject>

2. Interface IHavePriority should not request "Set" accessor, keep "Get" accessor is enough for it.

zhuxb711 commented:
@noamyogev84 Sorry to merge some improvements to this PR. It's perfect now.

noamyogev84 commented:
Thanks @zhuxb711 💪💪💪

zhuxb711 commented:
Hi @noamyogev84 if no concerns, please help merge this PR since I do not have permission to merge the PR. Thanks!

zhuxb711 commented:
Looking forward to our new nuget package!

noamyogev84 commented:
> Looking forward to our new nuget package!

https://www.nuget.org/packages/ConcurrentPriorityQueue/2.1.0
